### ISSUE TITLE
Mediod inside graph

### ISSUE THREAD

**Issue by AntonisZks** [AntonisZks/Information-Systems-Software-Development-Project#58]
I suggest we put the `findMediod()` function inside the **Vamana** class as a **method**. Additionally we can add an attribute to the class also, so that every time we load a vamana index from a file the **value** of the mediod will be loaded too. In that way we won't have to calculate the mediod every time we run the **Vamana**.

- [ ] Delete the `findMediod()` from the location it currently is.
- [ ] Since a new algorithm for the `findMediod()` is being given to us, I suggest we build the function according to that algorithm.
- [ ] Add the `findMediod()` inside the `VamanaIndex` class.
- [ ] Store the **mediod value** inside the file.

@EvitaChasioti Can you implement the function according to the provided algorithm from the paper?

**Comment by StavrosJKw** [comment]
I agree with your suggestions @AntonisZks, I will take the responsibility of implementing this, you can assign this job to me

**Comment by StavrosJKw** [comment]
- The `GraphNode<T> findMedoid(const Graph<T>& graph)` method is currently located in the directory `/include/vanama.h` in line 120. 
- The `VamanaIndex` class is located in  the directory `/include/Algorithms/VamanaIndex.h` in **line 74**.

**Comment by StavrosJKw** [comment]
I deleted and transferred the `FindMedoid()` function as a **method** inside the `VamanaIndex` class. 

I have not updated the algorithm yet with the one based on the new paper. I believe we should implement it first as a separate method like `FilteredMedoid` and after we have test it proceed to the deletion of the old `FindMedoid()`.
